pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1321,6 +1321,11 @@ type ObjectStoreSpec struct {
 	// +optional
 	// +nullable
 	Security *SecuritySpec `json:"security,omitempty"`
+
+	// +optional
+	// +kubebuilder:validation:Enum={"kms","s3","kms|s3"}
```

Comment:
Is `kms|s3` actually an option? On the linked doc it says these options are mutually exclusive.

Reply:
And if it is a valid option for rgw, shouldn't we leave them as independent bools per the previous discussion? Let's decide for sure on the design before updating the implementation.

Reply:
@travisn : for s3 clients sse options are mutually exclusive and it can send requests with only one of them `sse:c`,`sse:kms`,`sse:s3`. This option helps the RGW server configure with options so that `sse` requests from the client and setup connection with an external entity like `vault`. 
AFAIR for `sse:c` the involvement of external entity like `vault` is not required, encryptions are directly given by the user, so RGW just encrypts the data with that key

Reply:
I see, they are mutually exclusive for the clients, but RGW can allow any of the supported options. Then for the CRD schema I would still recommend the booleans for whether each one is enabled, instead of a single string value.

Reply:
@BlaineEXE @leseb ^^

Reply:
Okay. But how does this interact with the existing `Security` config above? Can both be used?

Also, I don't think an enum is the right way to go here. Why would we define all permutations of possibilities together and not have a struct where multiple can be enabled/disabled? This isn't a very good design for users or code.

Reply:
> You mention the `sse:c` option, but I don't see that here.

RGW does not take any specific configurations to support `sse:c`, hence it is not specified here

Reply:
> Okay. But how does this interact with the existing `Security` config above? Can both be used?

Security contains details about external services such as `vault`, so it is shared for both configs. 

> 
> Also, I don't think an enum is the right way to go here. Why would we define all permutations of possibilities together and not have a struct where multiple can be enabled/disabled? This isn't a very good design for users or code.